sound/soc/sof/sof-acpi-dev.c

```diff
@@ -232,17 +210,13 @@ static int sof_acpi_probe(struct platform_device *pdev)
 		if (ret < 0)
 			return ret;
 #else
-		dev_warn(dev, "No matching ASoC machine driver found - falling back to HDA codec\n");
-		mach = snd_soc_acpi_intel_hda_machines;
-		mach->sof_fw_filename = desc->nocodec_fw_filename;
+		dev_err(dev, "No matching ASoC machine driver found - aborting probe\n");
+		return -ENODEV;
```

Comment:
does this means there will never be soc HDA/HDMI support on ACPI probing platforms?

Reply:
never say never, right?
For now the direction is that all Intel devices are PCI-based. 
There are HDAudio-based devices without PCI (Nvidia) but I have no idea is they can even be supported with SOF.
The point is that for now this is dead code that can't be tested, so let's remove it.